services/blockchain-indexer/shared/dataService/delegates.js

```diff
@@ -0,0 +1,357 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const BluebirdPromise = require('bluebird');
+
+const { Logger, CacheRedis } = require('lisk-service-framework');
+const Transactions = require('@liskhq/lisk-transactions');
+
+const { getLastBlock } = require('./blocks');
+const dataService = require('./business');
+
+const {
+	getHexAddressFromBase32,
+	getBase32AddressFromPublicKey,
+	getBase32AddressFromHex,
+} = require('../utils/accountUtils');
+const { parseToJSONCompatObj } = require('../utils/parser');
+const requestAll = require('../utils/requestAll');
+const Signals = require('../utils/signals');
+const config = require('../../config');
+
+const cacheRedisDelegates = CacheRedis('delegates', config.endpoints.cache);
+
+const logger = Logger();
+
+const delegateStatus = {
+	ACTIVE: 'active',
+	STANDBY: 'standby',
+	BANNED: 'banned',
+	PUNISHED: 'punished',
+	NON_ELIGIBLE: 'non-eligible',
```

Comment:
Please add a TODO to update the value to 'ineligible' instead of 'non-eligible'
```suggestion
	NON_ELIGIBLE: 'non-eligible', // TODO: Update to 'ineligible' with API v3
```